app/utils.ts

```diff
@@ -0,0 +1,23 @@
+import { readJSON } from "fs-extra"
+import { PackageJson } from "gatsby"
+import path from "path"
+export async function hasGatsbyInstalled(root: string): Promise<boolean> {
+  try {
+    const packageJson = await readJSON(
+      path.join(root, `node_modules`, `gatsby`, `package.json`)
+    )
+    return packageJson?.name === `gatsby`
+  } catch (e) {
+    console.warn({ e })
+    return false
+  }
+}
+
+export async function loadPackageJson(root: string): Promise<PackageJson> {
+  return readJSON(path.join(root, `package.json`))
+}
+
+export function hasGatsbyDependency(packageJson: PackageJson): boolean {
+  const { dependencies = {}, devDependencies = {} } = packageJson
+  return !!{ ...dependencies, ...devDependencies }.gatsby
```

Comment:
Nit: Spread seems a bit overkill

```suggestion
  return !!dependencies.gatsby || !!devDependencies.gatsby
```

Or even (maybe ...)

```suggestion
  return !!packageJson.dependencies?.gatsby || !!packageJson.devDependencies?.gatsby
```